api/rosetta/payloads.go

```diff
@@ -0,0 +1,115 @@
+// the License.
+
+package rosetta
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/optakt/flow-dps/rosetta/failure"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/optakt/flow-dps/rosetta/object"
+)
+
+// PayloadsRequest implements the request schema for /construction/payloads.
+// See https://www.rosetta-api.org/docs/ConstructionApi.html#request-5
+type PayloadsRequest struct {
+	NetworkID  identifier.Network `json:"network_identifier"`
+	Operations []object.Operation `json:"operations"`
+	Metadata   object.Metadata    `json:"metadata"`
+}
+
+// PayloadsResponse implements the response schema for /construction/payloads.
+// See https://www.rosetta-api.org/docs/ConstructionApi.html#response-5
+type PayloadsResponse struct {
+	Transaction string                  `json:"unsigned_transaction"`
+	Payloads    []object.SigningPayload `json:"payloads"`
+}
+
+// Payloads implements the /construction/payloads endpoint of the Rosetta Construction API.
+// Payloads endpoint receives an array of operations and all other relevant information
+// required to construct an unsigned transaction. Operations must deterministically
+// describe the intent of the transaction. Besides the unsigned transaction text,
+// this endpoint also returns the list of payloads that should be signed.
+// See https://www.rosetta-api.org/docs/ConstructionApi.html#constructionpayloads
+func (c *Construction) Payloads(ctx echo.Context) error {
+
+	var req PayloadsRequest
+	err := ctx.Bind(&req)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidEncoding(invalidJSON, err))
+	}
+
+	if req.NetworkID.Blockchain == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(blockchainEmpty))
+	}
+	if req.NetworkID.Network == "" {
+		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(networkEmpty))
+	}
+
+	if len(req.Operations) != 2 {
+		return echo.NewHTTPError(http.StatusBadRequest,
+			invalidFormat(txInvalidOpCount,
+				withDetail("have_operations", len(req.Operations))),
+		)
+	}
+
+	intent, err := c.parser.CreateTransactionIntent(req.Operations)
+	var iaErr failure.InvalidAccount
+	if errors.As(err, &iaErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidAccount(iaErr))
+	}
+	var icErr failure.InvalidCurrency
+	if errors.As(err, &icErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidCurrency(icErr))
+	}
+	var ucErr failure.UnknownCurrency
+	if errors.As(err, &ucErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, unknownCurrency(ucErr))
+	}
+	var inErr failure.InvalidIntent
+	if errors.As(err, &inErr) {
+		return echo.NewHTTPError(http.StatusUnprocessableEntity, invalidIntent(inErr))
+	}
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, internal(intentDetermine, err))
+	}
+
+	tx, err := c.parser.CreateTransaction(intent)
```

Comment:
I think here we could work on the naming as well. `Create` is pretty generic, can we use something else? Well, we are turning something of one format to something of another format. So `ConvertTo`, or in this case I think I would like `Compile`.